packages/core/src/provider/base.ts

```diff
@@ -109,12 +80,17 @@ export class SmartAccountProvider<
     | PublicErc4337Client<TTransport>
     | PublicErc4337Client<HttpTransport>;
 
-  constructor({
-    rpcProvider,
-    entryPointAddress,
-    chain,
-    opts,
-  }: SmartAccountProviderConfig<TTransport>) {
+  constructor(config: SmartAccountProviderConfig<TTransport>) {
+    try {
+      SmartAccountProviderConfigSchema<TTransport | HttpTransport>().parse(
```

Comment:
this should be just `TTransport` right? not `HttpTransport`?

Reply:
oh yeah I was copying from the `rpcClient` field on the provider: 

```
  rpcClient:
    | PublicErc4337Client<TTransport>
    | PublicErc4337Client<HttpTransport>;
```

Reply:
but can remove HTTPTransport and it'll work the same